engine/cluster/src/main/java/org/opennms/oce/engine/cluster/ClusterEngine.java

```diff
@@ -373,28 +375,29 @@ public synchronized void onTick(long timestampInMillis) {
         if (alarmsBySituationId.size() == 1) {
             // All of the alarms in the cluster are not associated with a situation yet
             // Create a new situation with all of the alarms
-            final SituationBean situation = new SituationBean();
-            situation.setId(UUID.randomUUID().toString());
-            situation.setCreationTime(timestampInMillis);
+            final ImmutableSituation.Builder situationBuilder = ImmutableSituation.newBuilder()
+                    .setId(UUID.randomUUID().toString())
+                    .setCreationTime(timestampInMillis);
             for (AlarmInSpaceTime alarm : clusterOfAlarms.getPoints()) {
-                addAlarmToSituation(alarm.getAlarm(), situation);
+                situationBuilder.addAlarm(alarm.getAlarm(), this::isAlarmBlacklistedFromSituation);
             }
-            situations.add(situation);
+            situations.add(situationBuilder.build());
         } else if (alarmsBySituationId.size() == 2) {
             // Some of the alarms in the cluster already belong to a situation whereas other don't
             // Add them all to the same situation
             final String situationId = alarmsBySituationId.keySet().stream().filter(k -> !EMPTY_SITUATION_ID.equals(k))
                     .findFirst().orElseThrow(() -> new IllegalStateException("Should not happen."));
             // Create a copy of the existing situation
-            final SituationBean situation = new SituationBean(situationsById.get(situationId));
-            if (situation.getId() == null) {
+            final ImmutableSituation.Builder situationBuilder =
+                    ImmutableSituation.newBuilderFrom(situationsById.get(situationId));
+            if (situationBuilder.getId() == null) {
```

Comment:
This can't happen anymore, since the situation doesn't allow for null ids.

Reply:
Right, as long as we are using an impl that enforces that, its not possible anymore. Will remove.